### PR TITLE
Simplification of the Suffragium.sol contract + best practices

### PR DESCRIPTION
**Pull request by poppyseedDev**
#### Optimizations in `castVote`:
1. **Removed Redundant `allowTransient` Call:** 
   - The `allowTransient` call on the local contract is unnecessary, as the input was already validated with `asEuint64()` when created. This method is typically useful only when passing encrypted values as parameters to other contracts.

2. **Optimized `encryptedSupport` Type:**
   - Since `encryptedSupport` represents a binary choice (0 or 1), switching its input type to `ebool` enhances performance. It is then cast to `euint64` for tally purposes, reducing computational overhead.

#### Code Removal:
- **Removed `encryptedValidVotes`:**
   - This variable, which tracked the count of encrypted votes, has been removed to simplify the code. Total votes are already accessible via on-chain data, making this redundant.

### CODE REVIEW

File: contracts/contracts/Suffragium.sol [hunk 1]
```diff
@@ -103,11 +83,28 @@ contract Suffragium is ISuffragium, IdentityManager, GatewayCaller, Ownable {
         return _castedVotes[voteId][voterId];
     }
 
+    /// @notice Returns the number of votes cast for a specific vote
+    /// @param voteId ID of the vote to get the count for
+    /// @return The number of votes cast
+    function getVoteCount(uint256 voteId) external view returns (uint256) {
```
Comment (allemanfredi) [inline contracts/contracts/Suffragium.sol]: i would avoid to create this function as it's possible to simply call `vote(voteId)`

File: contracts/contracts/Suffragium.sol [hunk 2]
```diff
@@ -103,11 +83,28 @@ contract Suffragium is ISuffragium, IdentityManager, GatewayCaller, Ownable {
...
         if (vote.state != VoteState.Revealed) return false;
-        return (vote.result * 10 ** 18) / vote.validVotes >= vote.minQuorum;
+        if (vote.result == 0) return false;
+        return (vote.result * 10 ** 18) / vote.voteCount >= vote.minQuorum;
+    }
+
+    /// @notice Returns the result of a vote after it has been revealed
+    /// @param voteId ID of the vote to get the result for
+    /// @return The decrypted result of the vote
+    function getVoteResult(uint256 voteId) external view returns (uint256) {
```
Comment (allemanfredi) [inline contracts/contracts/Suffragium.sol]: same here

**Review by allemanfredi — CHANGES_REQUESTED** [review]
Just a couple of small changes. The changes within `castVote` make sense

File: contracts/contracts/Suffragium.sol (hunk 1 above)
Comment (poppyseedDev) [reply contracts/contracts/Suffragium.sol]: Sure can delete them, they were just helper functions for testing

File: contracts/contracts/Suffragium.sol [hunk 3]
```diff
@@ -103,11 +83,28 @@ contract Suffragium is ISuffragium, IdentityManager, GatewayCaller, Ownable {
         return _castedVotes[voteId][voterId];
     }
 
+    /// @notice Returns the number of votes cast for a specific vote
+    /// @param voteId ID of the vote to get the count for
+    /// @return The number of votes cast
+    function getVoteCount(uint256 voteId) external view returns (uint256) {
+        Vote storage vote = _getVote(voteId);
+        return vote.voteCount;
+    }
```
Comment (poppyseedDev) [inline contracts/contracts/Suffragium.sol]:
```suggestion

```

File: contracts/contracts/Suffragium.sol [hunk 4]
```diff
@@ -103,11 +83,28 @@ contract Suffragium is ISuffragium, IdentityManager, GatewayCaller, Ownable {
...
+        return (vote.result * 10 ** 18) / vote.voteCount >= vote.minQuorum;
+    }
+
+    /// @notice Returns the result of a vote after it has been revealed
+    /// @param voteId ID of the vote to get the result for
+    /// @return The decrypted result of the vote
+    function getVoteResult(uint256 voteId) external view returns (uint256) {
+        Vote storage vote = _getVote(voteId);
+        return vote.result;
     }
```
Comment (poppyseedDev) [inline contracts/contracts/Suffragium.sol]:
```suggestion

```

**Comment by poppyseedDev** [comment]
Tests are passing 👍

**Review by allemanfredi — APPROVED** [review]
LGTM! Thanks

**Pull request merged**